src/influence_diagram.jl

```diff
@@ -993,49 +972,173 @@ function generate_diagram!(diagram::InfluenceDiagram;
     positive_path_utility::Bool=false,
     negative_path_utility::Bool=false)
 
-
-    # Sort probabilities and consequences
-    sort!(diagram.X, by = x -> x.c)
-    sort!(diagram.Y, by = x -> x.v)
-
-
     # Declare P and U if defaults are used
     if default_probability
-        diagram.P = DefaultPathProbability(diagram.C, diagram.I_j[diagram.C], diagram.X)
+        C_indices = indices(diagram.C)
+        C_I_j_indices = I_j_indices(diagram, diagram.C)
+        diagram.P = DefaultPathProbability(C_indices, C_I_j_indices, get_values(diagram.X))
     end
+
     if default_utility
-        diagram.U = DefaultPathUtility(diagram.I_j[diagram.V], diagram.Y)
+        V_I_j_indices = I_j_indices(diagram, diagram.V)
+
+        diagram.U = DefaultPathUtility(V_I_j_indices, get_values(diagram.Y))
         if positive_path_utility
-            # Conversion to Float32 using Utility(), since machine default is Float64
-            diagram.translation = 1 -  minimum(diagram.U(s) for s in paths(diagram.S))
+            diagram.translation = 1 -  minimum(diagram.U(s) for s in paths(get_values(diagram.S)))
         elseif negative_path_utility
-            diagram.translation = -1 - maximum(diagram.U(s) for s in paths(diagram.S))
+            diagram.translation = -1 - maximum(diagram.U(s) for s in paths(get_values(diagram.S)))
         else
             diagram.translation = 0
         end
     end
+end
+
+
+"""
+    function indices(dict)
+
+Get the indices of nodes in values of a Dict or OrderedDict.
+
+# Example
+```julia-repl
+julia> D_indices = indices(diagram.D)
+3-element Vector{Int16}:
+ 3
+ 6
+ 9
+```
+"""
+#function indices(dict::OrderedDict{Any, Any}), MIKSEI TÄMÄ TOIMI, NYT TÄYTYY VAIN MÄÄRITELLÄ DICT ILMAN TYYPPIÄ JA TARVITAAN ERROR KÄSITTELIJÄ
```

Comment:
I tried defining a function with parameter dict::OrderedDict{Any, Any} and it seemed to work. What was the issue you were facing?

Reply:
Also, remove unnecessary comments